CONTRIBUTING.md

```diff
@@ -0,0 +1,290 @@
+makes it significantly easier to use or understand Kuma, congratulations!
+You are now an official contributor to Kuma. Get in touch with us to receive
+your very own [Contributor T-shirt](#contributor-t-shirt)!
+
+Your change will be included in the subsequent release Changelog, and we will
+not forget to include your name if you are an external contributor. :wink:
+
+[Back to TOC](#table-of-contents)
+
+#### Git branches
+
+We work on two branches: `master`, where non-breaking changes land, and 
+`feature/latest`, where important features or breaking changes land in-between
+major releases.
+
+When contributing to Kuma, this distinction is important. Please ensure that
+you are basing your work on top of the appropriate branch, it might save you
+some time down the road!
+
+If you have write access to the GitHub repository, please follow the following
+naming scheme when pushing your branch(es):
+
+- `feat/foo-bar` for new features
+- `fix/foo-bar` for bug fixes
+- `tests/foo-bar` when the change concerns only the test suite
+- `refactor/foo-bar` when refactoring code without any behavior change
+- `style/foo-bar` when addressing some style issue
+- `docs/foo-bar` for updates to the README.md, this file, or similar documents
+
+[Back to TOC](#table-of-contents)
+
+#### Commit atomicity
+
+When submitting patches, it is important that you organize your commits in
+logical units of work. You are free to propose a patch with one or many
+commits, as long as their atomicity is respected. This means that no unrelated
+changes should be included in a commit.
+
+For example: you are writing a patch to fix a bug, but in your endeavour, you
+spot another bug. **Do not fix both bugs in the same commit!** Finish your
+work on the initial bug, propose your patch, and come back to the second bug
+later on. This is also valid for unrelated style fixes, refactors, etc...
+
+You should use your best judgment when facing such decisions. A good approach
+for this is to put yourself in the shoes of the person who will review your
+patch: will they understand your changes and reasoning just by reading your
+commit history? Will they find unrelated changes in a particular commit? They
+shouldn't!
+
+Writing meaningful commit messages that follow our commit message format will
+also help you respect this mantra (see the below section).
+
+[Back to TOC](#table-of-contents)
+
+#### Commit message format
+
+To maintain a healthy Git history, we ask of you that you write your commit
+messages as follows:
+
+- The tense of your message must be **present**
+- Your message must be prefixed by a type, and a scope
+- The header of your message should not be longer than 50 characters
+- A blank line should be included between the header and the body
+- The body of your message should not contain lines longer than 72 characters
+
+Here is a template of what your commit message should look like:
+
+```
+<type>(<scope>) <subject>
```

Comment:
I like the defined types. Can we have a list of defined scopes?

Reply:
added following scopes:
- kuma-cp
- kuma-dp
- injector
- kumactp
- deps
- conf
- *